### PR TITLE
Ensure we create a new curl session if we need a new resolve_slist

### PR DESCRIPTION
**Pull request by jerryblakley**

### CODE REVIEW

**Comment by jerryblakley** [comment]
@kibra If we create a new resolve_slist, we don't actually put it into play until curl gets around to it (a couple of minutes or so, it seems). So if we remove a node, detect it, and create a new slist, curl will continue to route to the removed node for, I've seen, a minute and a half or so. During that time, due to a vaguary of our logging, we won't log events.
I've reordered some code to force a new session if we get a new slist. I've also left some REVIEW comments. We can force this new curl session to be created every time we get a new slist (every two minutes on every thread: performance/resource implications?) Or just when we detect a change to the platform (added/removed node or a change in a node's health status. Or we can create a new slist less often, e.g. every 10 minutes. If we see an error in those 10 minutes, we already and will continue to force a new slist and a new session.
Still testing. These changes are a little more involved than I'm comfortable just rolling out.
BTW, I also merged two smaller sets of changes to master earlier today.

File: src/session.c [hunk 1]
```diff
@@ -587,15 +591,24 @@ static int construct_resolve_slist(CURL *session, bool force) {
         filesystem_domain, filesystem_port);
     res = getaddrinfo(filesystem_domain, filesystem_port, &hints, &aihead);
```
Comment (kibra) [inline src/session.c]: It doesn't seem like we should call this `res` any more.

Maybe the next line should just be `if(getaddrinfo(filesystem_domain, filesystem_port, &hints, &aihead)) {`
?

**Comment by kibra** [comment]
@jerryblakley I don't think we should force a curl `handle_cleanup` every two minutes.

I would say we should only set `REQUIRES_NEW_SLIST` if the set of backends changes.

File: src/session.c (hunk 1 above)
Comment (jerryblakley) [inline src/session.c]: Turns out we query res to print the status of the error

**Pull request merged**